File: WpfApplication2/Polecenie.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WPFApplication2
{
    class Polecenie : System.Windows.Input.ICommand
    {
        public event EventHandler CanExecuteChanged;

        public bool CanExecute(object parameter)
        {
            return true;
        }

        public async void Execute(object parameter)
        {
            await Class1.Test();
            await Class1.Test();
        }
    }
}

File: WPFApplication3/Polecenie.pas
namespace WPFApplication3;

interface

type
  Polecenie = public class(System.Windows.Input.ICommand)
  private
  protected
  public
		event CanExecuteChanged: EventHandler;
		method Execute(parameter: Object);
		method CanExecute(parameter: Object): Boolean;
  end;

implementation

method Polecenie.Execute(parameter: Object);
begin
	await Class1.Test();
	await Class1.Test();
end;

method Polecenie.CanExecute(parameter: Object): Boolean;
begin
  result := true;
end;

end.
